src/Schema/Directives/CanDirective.php

```diff
@@ -95,7 +112,25 @@ function ($root, array $args, GraphQLContext $context, ResolveInfo $resolveInfo)
                 $gate = $this->gate->forUser($context->user());
                 $checkArguments = $this->buildCheckArguments($args);
 
-                foreach ($this->modelsToCheck($resolveInfo->argumentSet, $args) as $model) {
+                if ($this->directiveArgValue('findByArgs') === true) {
+                    if ($this->directiveArgValue('find') !== null) {
+                        throw new DefinitionException("Can not use the argument 'find' when 'findByArgs' is true");
+                    }
+                    $models = $resolveInfo
+                        ->argumentSet
+                        ->enhanceBuilder(
+                            $this->getModelClass()::query(),
+                            $this->directiveArgValue('scopes', [])
+                        )->get();
+                    if (! $models->count()) {
```

Comment:
I don't see why this check is necessary. An empty result is not necessarily an error.

Reply:
@spawnia The reason for this check is to keep the same behavior as the current methods. 
The `findOrFail()` will throw this exception.

Reply:
In order to avoid breakage, we should probably leave the existing behaviour as it is. In the case of `find` which only deals with primary keys, it is somewhat sensible to throw. Not finding a model by its primary key is usually indicative of an error.

This new argument does not mandate that the arguments have to look for a primary key, it can be any kind of constraint expressible through args and scopes. Empty results can be perfectly normal.